parla/task_runtime.py

```diff
@@ -1686,8 +1746,12 @@ def _construct_datamove_task(self, target_data, compute_task: ComputeTask, opera
                 if compute_task.is_dependent(dep_task):
                     if not datamove_task._add_dependency(dep_task):
                         completed_tasks.append(dep_task_id)
-            dep_task_list = [tuple(dt for dt in dep_task_list if dt[0] != ft) for ft in completed_tasks]
```

Comment:
Why do you remove this line?? It is view of dictionary and removes completed tasks.

Reply:
I didn't know what it was for! It seemed like an expensive loop that make something that was never used later. 

I don't think its a view. I think the assignment creates a new list and this new list is never used to update the dictionary.